cnd/src/storage.rs

```diff
@@ -56,22 +56,15 @@ impl Storage {
     }
 }
 
-#[async_trait::async_trait]
-impl Load<http_api::SwapContext> for Storage {
-    async fn load(&self, swap_id: LocalSwapId) -> anyhow::Result<http_api::SwapContext> {
-        self.db.load_swap_context(swap_id).await
-    }
-}
-
-#[async_trait::async_trait]
-impl Load<spawn::Swap<herc20::Params, halight::Params>> for Storage {
-    async fn load(
+impl Storage {
+    // Load swap for trading pair herc20/halight.
+    async fn load_tp_herc20_halight(
         &self,
         id: LocalSwapId,
-    ) -> anyhow::Result<spawn::Swap<herc20::Params, halight::Params>> {
+    ) -> anyhow::Result<(Swap, Herc20, Halight, Option<SecretHash>)> {
```

Comment:
Could you make `Herc20` and `Halight` generic here with a trait-bound to make the [`belonging_to`](https://docs.rs/diesel/1.4.5/diesel/prelude/trait.BelongingToDsl.html) method available?

Reply:
Even though it is just a private method, having a struct with field names would make it clearer that `.1` is `alpha` and `.2` is `beta`.

If we establish this and succeed in making this generic, this would be a nice place to do https://github.com/comit-network/comit-rs/issues/2781.

Reply:
As a follow-up, we can define traits for:

- derive_or_unwrap_secret_hash
- build_herc20_params
- build_halight_params
- build_hbit_params

and make one generic impl that loads `Params`, similar to what we did for `Save` !

Reply:
Can all be a follow-up so we don't make this one too big.

Reply:
> Even though it is just a private method, having a struct with field names would make it clearer that .1 is alpha and .2 is beta.

These are not sides (alpha/beta), this function returns the trading pair - it is agnostic to the side.